go/interpreter/lfvm/instructions_test.go

```diff
@@ -880,3 +880,60 @@ func TestExpansionCostOverflow(t *testing.T) {
 		}
 	}
 }
+
+func TestCallChargesAppropriatelyForColdWarmAccess(t *testing.T) {
+
+	tests := map[string]struct {
+		accessStatus tosca.AccessStatus
+		cost         tosca.Gas
+	}{
+		"warm": {
+			accessStatus: tosca.WarmAccess,
+		},
+		"cold": {
+			accessStatus: tosca.ColdAccess,
+			cost:         2500,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+
+			cost := getAccessCost(test.accessStatus)
+
+			if cost != test.cost {
+				t.Errorf("unexpected gas cost, wanted %v, got %v", test.cost, cost)
+			}
+
+		})
+	}
+}
```

Comment:
How about just this:
```
func TestGetAccessCost_RespondsWithProperGasPrice(t *testing.T) {
	if want, got := tosca.Gas(0), getAccessCost(tosca.WarmAccess); want != got {
		t.Errorf("unexpected gas cost, wanted %d, got %d", want, got)
	}
	if want, got := tosca.Gas(2500), getAccessCost(tosca.ColdAccess); want != got {
		t.Errorf("unexpected gas cost, wanted %d, got %d", want, got)
	}
}
```
or if a table should be needed, this
```
func TestGetAccessCost_RespondsWithProperGasPrice(t *testing.T) {
	prices := map[tosca.AccessStatus]tosca.Gas{
		tosca.WarmAccess: 0,
		tosca.ColdAccess: 2500,
	}
	for accessType, expectedPrice := range prices {
		if got := getAccessCost(accessType); got != expectedPrice {
			t.Errorf("unexpected gas cost, wanted %d, got %d", expectedPrice, got)
		}
	}
}
```

There is no need to stick to the table-driven testing formula with sub-tests if there is no need for it. Table-driven tests are good if there is some setup code needed or complex checks need to be performed, that could be re-used. The aim is to reduce code size and increase readability. But in examples like this, it actually adds complexity.